app/gui/view/component-browser/breadcrumbs/src/entry.rs

```diff
@@ -0,0 +1,343 @@
+//! A module defining the special [`Entry`] type for the grid view of the breadcrumbs.
+
+use ensogl_core::display::shape::*;
+use ensogl_core::prelude::*;
+
+use ensogl_core::application::command::FrpNetworkProvider;
+use ensogl_core::application::frp::API;
+use ensogl_core::application::Application;
+use ensogl_core::data::color;
+use ensogl_core::display;
+use ensogl_core::display::scene::Layer;
+use ensogl_core::Animation;
+use ensogl_grid_view::entry::Contour;
+use ensogl_grid_view::entry::EntryFrp;
+use ensogl_hardcoded_theme::application::component_browser::searcher::list_panel::breadcrumbs as theme;
+use ensogl_text as text;
+
+
+
+// ==============
+// === Shapes ===
+// ==============
+
+/// A filled triangle pointing to the right. Used to separate breadcrumbs.
+pub mod separator {
+    use super::*;
+    use std::f32::consts::PI;
+
+    pub const ICON_WIDTH: f32 = 16.0;
+
+    ensogl_core::define_shape_system! {
+        above = [ensogl_grid_view::entry::shape];
+        pointer_events = false;
+        (style: Style, color: Vector4) {
+            let width = style.get_number(theme::separator::width);
+            let height = style.get_number(theme::separator::height);
+            let triangle = Triangle(width, height).rotate((PI/2.0).radians());
+            let shape = triangle.fill(color);
+            shape.into()
+        }
+    }
+}
+
+/// A three dots icon. Used to indicate that the last module in breadcrumbs list contains more
+/// ancestors.
+pub mod ellipsis {
+    use super::*;
+
+    pub const ICON_WIDTH: f32 = 32.0;
+
+    ensogl_core::define_shape_system! {
+        above = [ensogl_grid_view::entry::shape];
+        pointer_events = false;
+        (style: Style, alpha: f32) {
+            let radius = style.get_number(theme::ellipsis::circles_radius).px();
+            let gap = style.get_number(theme::ellipsis::circles_gap).px();
+            let background_width = style.get_number(theme::ellipsis::background_width);
+            let background_height = style.get_number(theme::ellipsis::background_height);
+            let background_corners_radius = style.get_number(theme::ellipsis::background_corners_radius);
+            let col = style.get_color(theme::ellipsis::circles_color);
+            let circles_color = Var::<color::Rgba>::rgba(col.red,col.green,col.blue,alpha.clone());
+            let bg_col = style.get_color(theme::ellipsis::background_color);
+            let background_color = Var::<color::Rgba>::rgba(bg_col.red,bg_col.green,bg_col.blue,alpha);
+
+            let left = Circle(radius.clone()).fill(circles_color.clone());
+            let center = Circle(radius.clone()).fill(circles_color.clone());
+            let right = Circle(radius).fill(circles_color);
```

Comment:
Instead of creating 3 circles, use the repeat function (our drawing API). It is WAY faster. Repeat adds 1 opcode to GPU code to repeat the same element infinite amount of times.

Reply:
This does not explain anything. You told that you faced issue "similar to", whcih doesnt describe what issue you were facing. The comment in the code only tells "its broken", not explaining what is happening there. We are using intersections in several places, it's worth understanding what happens here. Please describe it properly here + attach screenshots.

Reply:
After your message, I checked again, with a slightly different code - and it worked. Now I can explain what happens. This code works:
```rust
let circles = Circle(...).repeat(...);
let mask = Rect(...);
let circles = circles.intersection(mask).fill(circles_color);
```

This one does not:

```rust
let circles = Circle(...).repeat(...).fill(circles_color);
let mask = Rect(...);
let circles = circles.intersection(mask);
```

The slight difference in placement of the `fill` breaks our renderer:

<img width="95" alt="Screenshot 2022-09-13 at 22 46 06" src="https://user-images.githubusercontent.com/6566674/189987814-6cda5e5f-23d8-4e39-8c29-f4654cd4cec2.png">
<img width="124" alt="Screenshot 2022-09-13 at 22 45 16" src="https://user-images.githubusercontent.com/6566674/189987817-f5607c4d-f73e-4dfd-9db3-2b67dde0b9d4.png">

Reply:
very interesting - please create an issue with these screenshots on pivotal tracker in the graph editor epic and assing to me.